gdb-server/src/parser/mod.rs

```diff
@@ -58,7 +61,9 @@ pub enum Packet {
     /// Packet `g`
     ReadGeneralRegister,
     /// Packet `G`
-    WriteGeneralRegister,
+    WriteGeneralRegister {
+        reg_values: Vec<u32>,
```

Comment:
The register can be different sizes, see for example the floating point registers for ARM:
https://github.com/probe-rs/probe-rs/blob/10a927cae43acb7b79c61bd52f9cbfddee5bfddb/gdb-server/src/architecture.rs#L35

I think the parse can't decode the register values, because they depend on the architecture. For now, it's probably better to parse the register values directly in the handler.

Reply:
There are some other places in the source, which assumes 32 bit registers.

For example the core register R/W methods: https://github.com/probe-rs/probe-rs/blob/10a927cae43acb7b79c61bd52f9cbfddee5bfddb/probe-rs/src/core/mod.rs#L161

The u32 should probably be replaced with a flexible bitvector like type.

Reply:
We can only read/write 32 bit values from the core, the corresponding register is only 32 bit wide (DCRDR, see C1.6.4 in the [ARMv7-M Architecture Reference](https://static.docs.arm.com/ddi0403/e/DDI0403E_d_armv7m_arm.pdf)).

From what I understand, the values for GDB are actually combined values, so we would need to read two registers, and then create a single 64 bit value out of that, or split the values when performing a write.

Anyway, I think the floating point registers are completely broken right now. We don't differentiate between the different FPUs, and we don't check if an FPU is present. This also plays in the support for target descriptions, we should indicate the presence of an FPU there. But that is stuff for another PR.

I think for this PR it's best if you could move the decoding to the handler, and implement the write the same as the read. We can then fix the register handling at a later point.